### PR TITLE
feat(cosmos): init client

### PR DESCRIPTION
**Pull request by frankcai4real**
## before you run unit test
- set up simapp

### CODE REVIEW

File: wallet/cosmos/cosmos.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	return a.client.Close()
+}
+
+func (a *WalletAdaptor) GetBalance(req *wallet.BalanceRequest) (*wallet.BalanceResponse, error) {
+	ret, err := a.client.GetBalance(context.Background(), req.Coin, req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	return &wallet.BalanceResponse{
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:46]: need reurn msg and code

File: wallet/cosmos/cosmos.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+					toAddr = attr.Value
+				}
+				if attr.Key == "sender" {
+					fromAddr = attr.Value
+				}
+			}
+		}
+	}
+
+	return &wallet.TxHashResponse{
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:81]: same as above

File: wallet/cosmos/cosmos.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	account := new(authv1beta1.BaseAccount)
+	if err := ptypes.UnmarshalAny(ret.Account, account); err != nil {
+		return nil, err
+	}
+
+	return &wallet.AccountResponse{
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:228]: same as above

File: wallet/cosmos/client.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+type Client struct {
+	grpcConn        *grpc.ClientConn
+	bankClient      banktypes.QueryClient
+	txServiceClient tx.ServiceClient
+	authClient      authv1beta1.QueryClient
+	bankKeeper      keeper.BaseKeeper
+}
+
+func NewClient(rpcTarget string) (*Client, error) {
```
Comment (guoshijiang) [inline wallet/cosmos/client.go:27]: use config for rpcTarget， like other chain, and should use client array for this NewClient

File: wallet/cosmos/cosmos.go [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+	}, nil
+}
+
+func (a *WalletAdaptor) GetGasPrice(req *wallet.GasPriceRequest) (*wallet.GasPriceResponse, error) {
+	// todo 怎么预估gasfee
+	return nil, nil
+}
+
+func (a *WalletAdaptor) SendTx(req *wallet.SendTxRequest) (*wallet.SendTxResponse, error) {
+	// a.client.SendTx(context.Background(), req.RawTx)
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:138]: This method should implement

File: wallet/cosmos/cosmos.go [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+		return nil, err
+	}
+
+	return &wallet.NonceResponse{
+		Nonce: strconv.FormatUint(account.GetSequence(), 10),
+	}, nil
+}
+
+func (a *WalletAdaptor) GetGasPrice(req *wallet.GasPriceRequest) (*wallet.GasPriceResponse, error) {
+	// todo 怎么预估gasfee
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:133]: This method should implement

File: wallet/cosmos/cosmos.go [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+}
+
+func (a *WalletAdaptor) SendTx(req *wallet.SendTxRequest) (*wallet.SendTxResponse, error) {
+	// a.client.SendTx(context.Background(), req.RawTx)
+	return nil, nil
+}
+
+func (a *WalletAdaptor) ConvertAddress(req *wallet.ConvertAddressRequest) (*wallet.ConvertAddressResponse, error) {
+	// todo
+	return nil, nil
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:144]: This method should implement

File: wallet/cosmos/cosmos.go [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+func (a *WalletAdaptor) ValidAddress(req *wallet.ValidAddressRequest) (*wallet.ValidAddressResponse, error) {
+	_, err := types2.AccAddressFromBech32(req.Address)
+	if err != nil {
+		return &wallet.ValidAddressResponse{
+			Code:  common.ReturnCode_ERROR,
+			Msg:   err.Error(),
+			Valid: false,
+		}, nil
+	}
+
```
Comment (guoshijiang) [inline wallet/cosmos/cosmos.go:156]: This method should implement address verify

File: wallet/cosmos/client.go (hunk 4 above)
Comment (frankcai4real) [reply wallet/cosmos/client.go:28]: done. use array outside the client, let the adaptor make the choice to select rpc server

**Pull request merged**